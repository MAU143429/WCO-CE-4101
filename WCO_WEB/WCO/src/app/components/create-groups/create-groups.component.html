<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="UTF-8">
    <!---Referencia al archivo css que contiene el estilo--->
    <link rel="stylesheet" href="create-groups.component.css">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
   </head>
<app-navbar></app-navbar>
<body>
  <div class="container">
    <div class="title">Crear un grupo</div>
    <div class="content">
      <form>
        <div class="group-details">

          <div class="input-box">
            <span class="details" >Nombre del grupo <span style="color: rgb(245, 59, 59);">*</span> </span>
            <input type="text"  [(ngModel)]= "newGroup.name"  [ngModelOptions]="{standalone: true}" minlength="5" maxlength="30"   placeholder="Ingrese un nombre de grupo..." required>
          </div>
          <div class="input-box">
            <span class="details" >Torneo<span style="color: rgb(245, 59, 59);">*</span></span>
            <mbsc-select [data]="allTournaments" [touchUi]="false"  [(ngModel)]= "newGroup.tId"  [ngModelOptions]="{standalone: true}"  placeholder="Seleccione el torneo" ></mbsc-select>
          </div>
        </div>
        <div class="button">
          <input type="submit" value="Crear Grupo" (click)="addGroup()">
        </div>
      </form>
    </div>
  </div>
</body>
</html>

