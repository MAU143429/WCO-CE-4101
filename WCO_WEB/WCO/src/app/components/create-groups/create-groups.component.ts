import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-create-groups',
  templateUrl: './create-groups.component.html',
  styleUrls: ['./create-groups.component.css'],
})
export class CreateGroupsComponent implements OnInit {
  allTournaments = [
    {
      text: 'Champions League #234352',
      value: 234352,
    },
  ];

  constructor() {}

  ngOnInit(): void {}
}
