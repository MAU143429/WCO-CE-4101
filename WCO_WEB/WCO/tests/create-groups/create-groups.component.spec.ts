import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CreateGroupsComponent } from '../../src/app/components/create-groups/create-groups.component';

describe('CreateGroupsComponent', () => {
  let component: CreateGroupsComponent;
  let fixture: ComponentFixture<CreateGroupsComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ CreateGroupsComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(CreateGroupsComponent);
    component = fixture.componentInstance;
    compiled = fixture.nativeElement;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hace match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });
});
