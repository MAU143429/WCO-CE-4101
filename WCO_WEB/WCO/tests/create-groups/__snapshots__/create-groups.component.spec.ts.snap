// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGroupsComponent debe de hace match con el snapshot 1`] = `"<html lang=\\"en\\" dir=\\"ltr\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"></head><app-navbar></app-navbar><body><div class=\\"container\\"><div class=\\"title\\">Crear un grupo</div><div class=\\"content\\"><form><div class=\\"group-details\\"><div class=\\"input-box\\"><span class=\\"details\\">Nombre del grupo <span style=\\"color: rgb(245, 59, 59);\\">*</span></span><input type=\\"text\\" minlength=\\"5\\" maxlength=\\"30\\" placeholder=\\"Ingrese un nombre de grupo...\\" required=\\"\\"></div><div class=\\"input-box\\"><span class=\\"details\\">Torneo<span style=\\"color: rgb(245, 59, 59);\\">*</span></span><mbsc-select placeholder=\\"Seleccione el torneo\\"></mbsc-select></div></div><div class=\\"button\\"><input type=\\"submit\\" value=\\"Crear Grupo\\"></div></form></div></div></body></html>"`;
